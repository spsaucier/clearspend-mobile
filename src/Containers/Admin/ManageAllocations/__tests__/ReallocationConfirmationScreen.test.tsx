import * as React from 'react';
import { fireEvent, cleanup, within } from '@testing-library/react-native';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { UserRolesAndPermissionsRecord } from '@/generated/capital';
import ReallocationConfirmationScreen from '../ReallocationConfirmationScreen';
import {
  ManageAllocationProvider,
  ReallocationType,
} from '@/Services/Admin/ManageAllocationProvider';
import { AdminScreens } from '@/Navigators/Admin/AdminNavigatorTypes';
import { adminResponse } from '@/Helpers/testing/fixtures/allocations';
import { bankAccounts } from '@/Helpers/testing/fixtures/business';
import { usersResponse } from '@/Helpers/testing/fixtures/user';
import { createQueryClient } from '@/Helpers/testing/reactQuery';
import { renderComponentWithQueryClient } from '@/Helpers/testing/WithQueryClient';

const mockedNavigate = jest.fn();

jest.mock('@react-navigation/core', () => ({
  useNavigation: () => ({
    navigate: mockedNavigate,
  }),
}));

const server = setupServer(
  rest.get(`/roles-and-permissions/allPermissions`, (req, res, ctx) =>
    res(ctx.json(adminResponse)),
  ),
  rest.get(`/business-bank-accounts`, (req, res, ctx) => res(ctx.json(bankAccounts))),
  rest.get(`/users`, (req, res, ctx) =>
    res(ctx.json(usersResponse.find((u) => u.type === 'BUSINESS_OWNER'))),
  ),
);

beforeAll(() => {
  server.listen();
});

afterEach(() => {
  cleanup();
  server.resetHandlers();
});

afterAll(() => {
  server.close();
});

const rootAllocation = adminResponse.allocations.find((a) => a.name === 'Dovetail');
const allocation = adminResponse.allocations.find((a) => a.name === 'Dinner');
const targetAllocation = adminResponse.allocations.find((a) => a.name === 'Breakfast');

describe('ReallocationConfirmationScreen', () => {
  it('renders', async () => {
    const { findByTestId } = renderComponentWithQueryClient(
      createQueryClient(),
      <ManageAllocationProvider
        allocationId={allocation?.allocationId!}
        reallocationType={ReallocationType.Add}
        targetAllocationId={targetAllocation?.allocationId!}
        allocations={adminResponse.allocations}
        userRoles={adminResponse.userRoles as UserRolesAndPermissionsRecord[]}
        userType="EMPLOYEE"
        amount="50"
      >
        <ReallocationConfirmationScreen />
      </ManageAllocationProvider>,
    );

    const testId = await findByTestId('manage-allocations-reallocation-confirmation-screen');
    expect(testId).toBeTruthy();

    const dinner = await findByTestId('Dinner-row');
    expect(within(dinner).queryByText('Dinner')).toBeTruthy();
    expect(within(dinner).queryByText('$108.00')).toBeTruthy();
    expect(dinner).toBeTruthy();

    const breakfast = await findByTestId('Breakfast-row');
    expect(within(breakfast).queryByText('Breakfast')).toBeTruthy();
    expect(within(breakfast).queryByText('$200.00')).toBeTruthy();
    expect(breakfast).toBeTruthy();
  });

  it('navigates to "Admin Dashboard" screen when selecting primary button', async () => {
    const { findByTestId } = renderComponentWithQueryClient(
      createQueryClient(),
      <ManageAllocationProvider
        allocationId={allocation?.allocationId!}
        reallocationType={ReallocationType.Add}
        targetAllocationId={targetAllocation?.allocationId!}
        allocations={adminResponse.allocations}
        userRoles={adminResponse.userRoles as UserRolesAndPermissionsRecord[]}
        userType="EMPLOYEE"
        amount="50"
      >
        <ReallocationConfirmationScreen />
      </ManageAllocationProvider>,
    );

    const primaryActionButton = await findByTestId('primary-action-button');

    fireEvent.press(primaryActionButton);

    expect(mockedNavigate).toHaveBeenCalledWith(AdminScreens.Home);
  });

  it('matches snapshot for "Rellocation" with "Add" funds', async () => {
    const result = renderComponentWithQueryClient(
      createQueryClient(),
      <ManageAllocationProvider
        allocationId={allocation?.allocationId!}
        reallocationType={ReallocationType.Add}
        targetAllocationId={targetAllocation?.allocationId}
        allocations={adminResponse.allocations}
        userRoles={adminResponse.userRoles as UserRolesAndPermissionsRecord[]}
        userType="EMPLOYEE"
      >
        <ReallocationConfirmationScreen />
      </ManageAllocationProvider>,
    );

    const summary = await result.findByTestId('confirmation-summary');

    expect(summary).toBeTruthy();
    expect(result).toMatchSnapshot();
  });

  it('matches snapshot for "Rellocation" with "Remove" funds', async () => {
    const result = renderComponentWithQueryClient(
      createQueryClient(),
      <ManageAllocationProvider
        allocationId={allocation?.allocationId!}
        reallocationType={ReallocationType.Remove}
        targetAllocationId={targetAllocation?.allocationId}
        allocations={adminResponse.allocations}
        userRoles={adminResponse.userRoles as UserRolesAndPermissionsRecord[]}
        userType="EMPLOYEE"
      >
        <ReallocationConfirmationScreen />
      </ManageAllocationProvider>,
    );

    const summary = await result.findByTestId('confirmation-summary');

    expect(summary).toBeTruthy();
    expect(result).toMatchSnapshot();
  });

  it('matches snapshot for "BankTransfer" with "Add" funds', async () => {
    const result = renderComponentWithQueryClient(
      createQueryClient(),
      <ManageAllocationProvider
        allocationId={rootAllocation?.allocationId!}
        reallocationType={ReallocationType.Add}
        targetAllocationId={bankAccounts[0].businessBankAccountId}
        allocations={adminResponse.allocations}
        userRoles={adminResponse.userRoles as UserRolesAndPermissionsRecord[]}
        userType="BUSINESS_OWNER"
      >
        <ReallocationConfirmationScreen />
      </ManageAllocationProvider>,
    );

    const summary = await result.findByTestId('confirmation-summary');

    expect(summary).toBeTruthy();
    expect(result).toMatchSnapshot();
  });

  it('matches snapshot for "BankTransfer" with "Remove" funds', async () => {
    const result = renderComponentWithQueryClient(
      createQueryClient(),
      <ManageAllocationProvider
        allocationId={rootAllocation?.allocationId!}
        reallocationType={ReallocationType.Remove}
        targetAllocationId={bankAccounts[0].businessBankAccountId}
        allocations={adminResponse.allocations}
        userRoles={adminResponse.userRoles as UserRolesAndPermissionsRecord[]}
        userType="BUSINESS_OWNER"
      >
        <ReallocationConfirmationScreen />
      </ManageAllocationProvider>,
    );

    const summary = await result.findByTestId('confirmation-summary');

    expect(summary).toBeTruthy();
    expect(result).toMatchSnapshot();
  });
});
