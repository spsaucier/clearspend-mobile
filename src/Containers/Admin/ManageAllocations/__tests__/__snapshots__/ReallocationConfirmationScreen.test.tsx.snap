// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReallocationConfirmationScreen matches snapshot for "BankTransfer" with "Add" funds 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "top",
      "bottom",
    ]
  }
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
  testID="manage-allocations-reallocation-confirmation-screen"
>
  <View
    style={
      Object {
        "marginBottom": 16,
        "marginTop": 96,
      }
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "PP Neue Montreal TT",
            "fontSize": 16,
          },
          Object {
            "color": "#000000",
            "fontFamily": "PP Telegraf",
            "fontSize": 24,
            "fontWeight": "300",
          },
        ]
      }
      testID="title"
    >
      adminFlows.manageAllocation.confirmationTitle
    </Text>
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "PP Neue Montreal TT",
            "fontSize": 16,
          },
          Object {
            "color": "#3F3F3F",
            "fontSize": 14,
            "lineHeight": 21,
            "marginBottom": 32,
            "marginTop": 16,
          },
        ]
      }
      testID="text"
    />
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "PP Neue Montreal TT",
            "fontSize": 16,
          },
          Object {
            "fontSize": 10,
            "fontWeight": "500",
            "letterSpacing": 1,
            "marginBottom": 8,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      adminFlows.manageAllocation.updatedBalances
    </Text>
    <View
      testID="confirmation-summary"
    >
      <View
        style={
          Object {
            "backgroundColor": "#F3F2EF",
            "marginBottom": 4,
            "paddingBottom": 12,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 12,
          }
        }
        testID="Dovetail-row"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            Dovetail
          </Text>
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            $708.00
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#F3F2EF",
            "marginBottom": 4,
            "paddingBottom": 12,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 12,
          }
        }
        testID="Plaid Saving-row"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "aspectRatio": 1,
                  "height": 20,
                  "width": 20,
                },
                undefined,
              ]
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="100%"
              bbWidth="100%"
              fill="none"
              focusable={false}
              height="100%"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
              width="100%"
            >
              <RNSVGGroup
                fill={null}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21Z"
                  fill={4278190080}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  clipRule={0}
                  d="M12 6.9375C12.3107 6.9375 12.5625 7.18934 12.5625 7.5V12.75C12.5625 13.0607 12.3107 13.3125 12 13.3125C11.6893 13.3125 11.4375 13.0607 11.4375 12.75V7.5C11.4375 7.18934 11.6893 6.9375 12 6.9375Z"
                  fill={4294967295}
                  fillRule={0}
                  propList={
                    Array [
                      "fill",
                      "fillRule",
                    ]
                  }
                />
                <RNSVGPath
                  d="M12 17.0625C12.5178 17.0625 12.9375 16.6428 12.9375 16.125C12.9375 15.6072 12.5178 15.1875 12 15.1875C11.4822 15.1875 11.0625 15.6072 11.0625 16.125C11.0625 16.6428 11.4822 17.0625 12 17.0625Z"
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 12,
                  "marginLeft": 8,
                },
              ]
            }
          >
            adminFlows.manageAllocation.bankTransferConfirmationDisclaimer
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": "auto",
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#43FA76",
          "borderRadius": 4,
          "display": "flex",
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
          "width": "100%",
        }
      }
      testID="primary-action-button"
    >
      <Text
        allowFontScaling={true}
        maxFontSizeMultiplier={1.5}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "PP Neue Montreal TT",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#003333",
              },
              undefined,
            ],
          ]
        }
      >
        adminFlows.manageAllocation.confirmationPrimaryActionCta
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`ReallocationConfirmationScreen matches snapshot for "BankTransfer" with "Remove" funds 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "top",
      "bottom",
    ]
  }
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
  testID="manage-allocations-reallocation-confirmation-screen"
>
  <View
    style={
      Object {
        "marginBottom": 16,
        "marginTop": 96,
      }
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "PP Neue Montreal TT",
            "fontSize": 16,
          },
          Object {
            "color": "#000000",
            "fontFamily": "PP Telegraf",
            "fontSize": 24,
            "fontWeight": "300",
          },
        ]
      }
      testID="title"
    >
      adminFlows.manageAllocation.confirmationTitle
    </Text>
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "PP Neue Montreal TT",
            "fontSize": 16,
          },
          Object {
            "color": "#3F3F3F",
            "fontSize": 14,
            "lineHeight": 21,
            "marginBottom": 32,
            "marginTop": 16,
          },
        ]
      }
      testID="text"
    />
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "PP Neue Montreal TT",
            "fontSize": 16,
          },
          Object {
            "fontSize": 10,
            "fontWeight": "500",
            "letterSpacing": 1,
            "marginBottom": 8,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      adminFlows.manageAllocation.updatedBalances
    </Text>
    <View
      testID="confirmation-summary"
    >
      <View
        style={
          Object {
            "backgroundColor": "#F3F2EF",
            "marginBottom": 4,
            "paddingBottom": 12,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 12,
          }
        }
        testID="Dovetail-row"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            Dovetail
          </Text>
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            $708.00
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#F3F2EF",
            "marginBottom": 4,
            "paddingBottom": 12,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 12,
          }
        }
        testID="Plaid Saving-row"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "aspectRatio": 1,
                  "height": 20,
                  "width": 20,
                },
                undefined,
              ]
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight="100%"
              bbWidth="100%"
              fill="none"
              focusable={false}
              height="100%"
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": "100%",
                    "width": "100%",
                  },
                ]
              }
              vbHeight={24}
              vbWidth={24}
              width="100%"
            >
              <RNSVGGroup
                fill={null}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21Z"
                  fill={4278190080}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
                <RNSVGPath
                  clipRule={0}
                  d="M12 6.9375C12.3107 6.9375 12.5625 7.18934 12.5625 7.5V12.75C12.5625 13.0607 12.3107 13.3125 12 13.3125C11.6893 13.3125 11.4375 13.0607 11.4375 12.75V7.5C11.4375 7.18934 11.6893 6.9375 12 6.9375Z"
                  fill={4294967295}
                  fillRule={0}
                  propList={
                    Array [
                      "fill",
                      "fillRule",
                    ]
                  }
                />
                <RNSVGPath
                  d="M12 17.0625C12.5178 17.0625 12.9375 16.6428 12.9375 16.125C12.9375 15.6072 12.5178 15.1875 12 15.1875C11.4822 15.1875 11.0625 15.6072 11.0625 16.125C11.0625 16.6428 11.4822 17.0625 12 17.0625Z"
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 12,
                  "marginLeft": 8,
                },
              ]
            }
          >
            adminFlows.manageAllocation.bankTransferConfirmationDisclaimer
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": "auto",
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#43FA76",
          "borderRadius": 4,
          "display": "flex",
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
          "width": "100%",
        }
      }
      testID="primary-action-button"
    >
      <Text
        allowFontScaling={true}
        maxFontSizeMultiplier={1.5}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "PP Neue Montreal TT",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#003333",
              },
              undefined,
            ],
          ]
        }
      >
        adminFlows.manageAllocation.confirmationPrimaryActionCta
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`ReallocationConfirmationScreen matches snapshot for "Rellocation" with "Add" funds 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "top",
      "bottom",
    ]
  }
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
  testID="manage-allocations-reallocation-confirmation-screen"
>
  <View
    style={
      Object {
        "marginBottom": 16,
        "marginTop": 96,
      }
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "PP Neue Montreal TT",
            "fontSize": 16,
          },
          Object {
            "color": "#000000",
            "fontFamily": "PP Telegraf",
            "fontSize": 24,
            "fontWeight": "300",
          },
        ]
      }
      testID="title"
    >
      adminFlows.manageAllocation.confirmationTitle
    </Text>
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "PP Neue Montreal TT",
            "fontSize": 16,
          },
          Object {
            "color": "#3F3F3F",
            "fontSize": 14,
            "lineHeight": 21,
            "marginBottom": 32,
            "marginTop": 16,
          },
        ]
      }
      testID="text"
    />
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "PP Neue Montreal TT",
            "fontSize": 16,
          },
          Object {
            "fontSize": 10,
            "fontWeight": "500",
            "letterSpacing": 1,
            "marginBottom": 8,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      adminFlows.manageAllocation.updatedBalances
    </Text>
    <View
      testID="confirmation-summary"
    >
      <View
        style={
          Object {
            "backgroundColor": "#F3F2EF",
            "marginBottom": 4,
            "paddingBottom": 12,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 12,
          }
        }
        testID="Breakfast-row"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            Breakfast
          </Text>
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            $200.00
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#F3F2EF",
            "marginBottom": 4,
            "paddingBottom": 12,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 12,
          }
        }
        testID="Dinner-row"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            Dinner
          </Text>
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            $108.00
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": "auto",
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#43FA76",
          "borderRadius": 4,
          "display": "flex",
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
          "width": "100%",
        }
      }
      testID="primary-action-button"
    >
      <Text
        allowFontScaling={true}
        maxFontSizeMultiplier={1.5}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "PP Neue Montreal TT",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#003333",
              },
              undefined,
            ],
          ]
        }
      >
        adminFlows.manageAllocation.confirmationPrimaryActionCta
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`ReallocationConfirmationScreen matches snapshot for "Rellocation" with "Remove" funds 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "top",
      "bottom",
    ]
  }
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
  testID="manage-allocations-reallocation-confirmation-screen"
>
  <View
    style={
      Object {
        "marginBottom": 16,
        "marginTop": 96,
      }
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "PP Neue Montreal TT",
            "fontSize": 16,
          },
          Object {
            "color": "#000000",
            "fontFamily": "PP Telegraf",
            "fontSize": 24,
            "fontWeight": "300",
          },
        ]
      }
      testID="title"
    >
      adminFlows.manageAllocation.confirmationTitle
    </Text>
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "PP Neue Montreal TT",
            "fontSize": 16,
          },
          Object {
            "color": "#3F3F3F",
            "fontSize": 14,
            "lineHeight": 21,
            "marginBottom": 32,
            "marginTop": 16,
          },
        ]
      }
      testID="text"
    />
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.5}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "PP Neue Montreal TT",
            "fontSize": 16,
          },
          Object {
            "fontSize": 10,
            "fontWeight": "500",
            "letterSpacing": 1,
            "marginBottom": 8,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      adminFlows.manageAllocation.updatedBalances
    </Text>
    <View
      testID="confirmation-summary"
    >
      <View
        style={
          Object {
            "backgroundColor": "#F3F2EF",
            "marginBottom": 4,
            "paddingBottom": 12,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 12,
          }
        }
        testID="Dinner-row"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            Dinner
          </Text>
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            $108.00
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#F3F2EF",
            "marginBottom": 4,
            "paddingBottom": 12,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 12,
          }
        }
        testID="Breakfast-row"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            Breakfast
          </Text>
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.5}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "PP Neue Montreal TT",
                  "fontSize": 16,
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            $200.00
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": "auto",
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#43FA76",
          "borderRadius": 4,
          "display": "flex",
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
          "width": "100%",
        }
      }
      testID="primary-action-button"
    >
      <Text
        allowFontScaling={true}
        maxFontSizeMultiplier={1.5}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "PP Neue Montreal TT",
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#003333",
              },
              undefined,
            ],
          ]
        }
      >
        adminFlows.manageAllocation.confirmationPrimaryActionCta
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
